### PR TITLE
Check transaction events for errors

### PR DESCRIPTION
**Pull request by coreyar**
Check and throw errors if a transaction event fails.
Errors are decoded using enums declared in the contracts - https://github.com/VenusProtocol/venus-protocol/blob/develop/contracts/ErrorReporter.sol
We'll need to update our errors when this enum changes because it is indexed by order

### CODE REVIEW

File: src/utilities/errors.ts [hunk 1]
```diff
@@ -21,3 +31,53 @@ export class InternalError extends Error {
...
+
+export const checkForTokenTransactionError = (receipt: TransactionReceipt) =>
+  checkForTransactionError(receipt, TokenErrorReporterError, TokenErrorReporterFailureInfo);
+
+export const checkForVaiControllerTransactionError = (receipt: TransactionReceipt) =>
+  checkForTransactionError(
+    receipt,
+    VAIControllerErrorReporterError,
+    VAIControllerErrorReporterFailureInfo,
+  );
```
Comment (therealemjy) [inline src/utilities/errors.ts:83]: Why not use `checkForTransactionError` directly if only the type changes?

**Review by therealemjy — APPROVED** [review]
Really nice work 👍
Just one minor comment

File: src/utilities/errors.ts (hunk 1 above)
Comment (coreyar) [reply src/utilities/errors.ts:83]: It is actually 2 and could be 3. The error is returned with 3 arguments that differ by contract `error`, `info` and `detail`. Rather have import 3 or four and risk mixing them up when using `checkForTransactionError` this seemed more convenient

Comment (therealemjy) [reply src/utilities/errors.ts:83]: Ok got you 👍

**Pull request merged**